pkg/api/vm/handler.go

```diff
@@ -633,15 +686,53 @@ func (h *vmActionHandler) removeVolume(ctx context.Context, namespace, name stri
 		return fmt.Errorf("failed to serialize payload,: %v", err)
 	}
 	// Ref: https://kubevirt.io/api-reference/v0.44.0/operations.html#_v1vm-removevolume
-	return h.virtSubresourceRestClient.
+	if err = h.virtSubresourceRestClient.
 		Put().
 		Namespace(namespace).
 		Resource(vmResource).
 		Name(name).
 		SubResource(strings.ToLower(removeVolume)).
 		Body(body).
 		Do(ctx).
-		Error()
+		Error(); err != nil {
+		return err
+	}
+
+	// Get VM object again, because we've updated it.
+	vm, err = h.vmCache.Get(namespace, name)
+	if err != nil {
+		return err
+	}
+
+	if pvcName != "" {
+		var volumeClaimTemplates []corev1.PersistentVolumeClaim
+		vmCopy := vm.DeepCopy()
+		anno := vmCopy.GetAnnotations()
+		if volumeClaimTemplatesJSON, ok := anno[util.AnnotationVolumeClaimTemplates]; ok {
+			if err = json.Unmarshal([]byte(volumeClaimTemplatesJSON), &volumeClaimTemplates); err != nil {
+				return fmt.Errorf("failed to unserialize %s, error: %v", util.AnnotationVolumeClaimTemplates, err)
+			}
+		}
+		for i, volumeClaimTemplate := range volumeClaimTemplates {
+			if volumeClaimTemplate.Name == pvcName {
+				volumeClaimTemplates = append(volumeClaimTemplates[:i], volumeClaimTemplates[i+1:]...)
```

Comment:
prefer to erase the element without keeping its order since the json.Marshal will not preserve it, and the order is not valuable to the VM.

Reply:
Updated it. Thanks.